### PR TITLE
Add Command and Handler Generators

### PR DESCRIPTION
**Pull request by timbroder**
I've been using this package a lot and wanted to make creating Commands and Handlers easier
## Generators

You can generate Commands and Handlers automatically using artisan

```
artisan make:tactician:command Foo
artisan make:tactician:handler Foo
```

This will create FooCommand and FooHandler and place them in the app/CommandBus/Commands and app/CommandBus/Handlers respectively

To run both at once

```
artisan make:tactician Foo
```

### CODE REVIEW

**Comment by timbroder** [comment]
@joselfonseca I've fixed the CI issues on this PR. 

Outstanding:
1. I believe the camel case issues in the Tests from Nitpick are false positives, as I followed your format
2. Nitpick and Scrutinizer disagree on this point, please let me know [which](https://scrutinizer-ci.com/g/joselfonseca/laravel-tactician/inspections/3e5b177d-ceec-4836-b283-a6accb8b16ae/patches?status=new) you'd like me to use

File: src/Providers/LaravelTacticianServiceProvider.php [hunk 1]
```diff
@@ -23,6 +26,24 @@ public function register()
         $this->app->bind('League\Tactician\Handler\MethodNameInflector\MethodNameInflector', config('laravel-tactician.inflector'));
         $this->app->bind('League\Tactician\Handler\CommandNameExtractor\CommandNameExtractor', config('laravel-tactician.extractor'));
         $this->app->bind('Joselfonseca\LaravelTactician\CommandBusInterface', config('laravel-tactician.bus'));
+
+        // Register Command Generator
+        $this->app->singleton('laravel-tactician.make.command', function($app) {
```
Comment (timbroder) [inline src/Providers/LaravelTacticianServiceProvider.php]: fixed in 762bf3f

File: src/Providers/LaravelTacticianServiceProvider.php [hunk 2]
```diff
@@ -23,6 +26,24 @@ public function register()
...
+            return new MakeTacticianCommandCommand($app['files']);
+        });
+        $this->commands('laravel-tactician.make.command');
+
+        // Register Handler Generator
+        $this->app->singleton('laravel-tactician.make.handler', function($app) {
```
Comment (timbroder) [inline src/Providers/LaravelTacticianServiceProvider.php]: fixed in 762bf3f

File: src/Providers/LaravelTacticianServiceProvider.php [hunk 3]
```diff
@@ -23,6 +26,24 @@ public function register()
...
+            return new MakeTacticianHandlerCommand($app['files']);
+        });
+        $this->commands('laravel-tactician.make.handler');
+
+        // Register Comman+Handler Generator Command
+        $this->app->singleton('laravel-tactician.make.tactician', function($app) {
```
Comment (timbroder) [inline src/Providers/LaravelTacticianServiceProvider.php]: fixed in 762bf3f

File: tests/Generators/TestGenerators.php [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+namespace Joselfonseca\LaravelTactician\Tests\Generators;
+
+use Illuminate\Support\Facades\Artisan;
+use Joselfonseca\LaravelTactician\Tests\TestCase;
+
+/**
+ * Class TestGenerators
+ * @package Joselfonseca\LaravelTactician\Tests\Generators
+ */
+class TestGenerators extends TestCase {
```
Comment (timbroder) [inline tests/Generators/TestGenerators.php]: fixed in 762bf3f

File: tests/Generators/TestGenerators.php [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+
+        if (file_exists($this->expectedHandlerFile)) {
+            unlink($this->expectedHandlerFile);
+        }
+    }
+
+    /**
+     * Generate Command using Artisan
+     */
+    protected function makeCommand() {
```
Comment (timbroder) [inline tests/Generators/TestGenerators.php]: fixed in 762bf3f

File: tests/Generators/TestGenerators.php [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        Artisan::call('make:tactician:command', ['name' => 'Foo']);
+    }
+
+    /**
+     * Generate Handler using Artisan
+     */
+    protected function makeHandler() {
```
Comment (timbroder) [inline tests/Generators/TestGenerators.php]: fixed in 762bf3f

**Comment by joselfonseca** [comment]
Hi @timbroder. Thanks for this great contribution! in regards of your questions, 
1. They are false positives as i like to use snake case for tests so it is easier to read.
2. Please use nitpick's. 

Thanks again and please let me know when i can merge.

**Comment by timbroder** [comment]
cool I do that too!

This is ready to merge

**Pull request merged**